### PR TITLE
DIV-6816: changes to direct petitioner to respondent solictor page

### PR DESCRIPTION
**Pull request by Dizrupto**
# Description

Add "Send to their solicitor" option when choosing respondent solicitor contact address.

Fixes https://tools.hmcts.net/jira/browse/DIV-6816

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by johnoneill46 — APPROVED** [review]

File: app/steps/respondent/correspondence/use-home-address/template.html [hunk 1]
```diff
@@ -1,10 +1,50 @@
 {% from "common/components/formElements.html" import radioButton %}
 {% extends "question.html" %}
 {% block question %}
-    {{ content.question | safe }}
+    {% if session.featureToggles.ft_represented_respondent_journey == true %}
```
Comment (madjava) [inline app/steps/respondent/correspondence/use-home-address/template.html:4]: @Dizrupto Could we not use this `ft_represented_respondent_journey` directly this way. If for any reason that changes then this break. Maybe have a variable set like we did for loading downloadable files and use that instead. That way this screen does not to be tighly coupled with this toggle

File: app/steps/respondent/correspondence/send-to-solicitor/content.json [hunk 2]
```diff
@@ -5,7 +5,13 @@
       "translation": {
         "content": {
           "question": "Where should your {{ divorceWho }}’s divorce papers be sent?",
-          "correspondence": "To a different address"
+          "correspondence": "To a different address",
+          "ft_resp_sol": {
```
Comment (madjava) [inline app/steps/respondent/correspondence/send-to-solicitor/content.json:9]: Might be better to have a less cryptic name that also follows the camel case convention 😜

File: app/steps/respondent/correspondence/send-to-solicitor/template.html [hunk 3]
```diff
@@ -5,20 +5,54 @@
 {% endblock %}
 
 {% block form %}
- <div class="govuk-form-group {{ 'govuk-form-group--error' if fields.respondentSolicitorRepresented.error }}">
-    <fieldset id="respondentSolicitorRepresented" class="govuk-fieldset">
+
+  {% if session.featureToggles.ft_represented_respondent_journey == true %}
```
Comment (madjava) [inline app/steps/respondent/correspondence/send-to-solicitor/template.html:9]: as mentioned in another comment, `session.featureToggles.ft_represented_respondent_journey` couples too tightly with this screen

File: config/default.yaml [hunk 4]
```diff
@@ -201,7 +201,7 @@ featureToggles:
   ft_pcq: 'petitioner-pcq'
   ft_awaiting_amend: 'divorce-amended-application'
   ft_represented_respondent_journey: 'represented-respondent-journey'
-  appwideToggles: ['ft_welsh']
+  appwideToggles: ['ft_welsh', 'ft_represented_respondent_journey']
```
Comment (madjava) [inline config/default.yaml:204]: Do we need his here as well?

File: app/steps/respondent/correspondence/use-home-address/template.html [hunk 5]
```diff
@@ -1,10 +1,50 @@
...
+        {{ content.ft_resp_sol.question | safe }}
+    {% else %}
+        {{ content.question | safe }}
+    {% endif %}
 {% endblock %}
 
 {% block form %}
+    {% if session.featureToggles.ft_represented_respondent_journey == true %}
```
Comment (madjava) [inline app/steps/respondent/correspondence/use-home-address/template.html:12]: Ditto

File: app/steps/respondent/correspondence/use-home-address/template.html (hunk 1 above)
Comment (Dizrupto) [reply app/steps/respondent/correspondence/use-home-address/template.html:4]: Hi Felix, yep good idea I will try that :)

File: app/steps/respondent/correspondence/send-to-solicitor/content.json (hunk 2 above)
Comment (Dizrupto) [reply app/steps/respondent/correspondence/send-to-solicitor/content.json:9]: aha okay yep sure :)

File: config/default.yaml (hunk 4 above)
Comment (Dizrupto) [reply config/default.yaml:204]: Yep to be able to use it from HTML :)

Comment (madjava) [reply config/default.yaml:204]: 10-4

**Review by madjava — APPROVED** [review]

**Review by yarpo — APPROVED** [review]

**Pull request merged**